commons/src/main/java/eu/itesla_project/commons/config/PlatformConfig.java

```diff
@@ -91,9 +83,84 @@ public static synchronized CacheManager defaultCacheManager() {
         return defaultCacheManager;
     }
 
+    protected PlatformConfig(FileSystem fileSystem) {
+        this(fileSystem, getDefaultConfigDir(fileSystem), getDefaultCacheDir(fileSystem));
+    }
+
+    protected PlatformConfig(FileSystem fileSystem, Path configDir, Path cacheDir) {
+        this.fileSystem = Objects.requireNonNull(fileSystem);
+        this.configDir = createConfigDir(configDir);
+        this.cacheDir = createCacheDir(cacheDir);
+    }
+
+    public FileSystem getFileSystem() {
+        return fileSystem;
+    }
+
+    public Path getConfigDir() {
+        return configDir;
+    }
+
+    public Path getCacheDir() {
+        return cacheDir;
+    }
+
     public abstract boolean moduleExists(String name);
 
     public abstract ModuleConfig getModuleConfig(String name);
 
     public abstract ModuleConfig getModuleConfigIfExists(String name);
+
+    static Path getDefaultConfigDir(FileSystem fileSystem) {
```

Comment:
getDefaultConfigDir and getDefaultCacheDir are very similar. Use of a common method to avoid duplication?